### ISSUE TITLE
Stuck IPv6 connections with Mullvad exit nodes

### ISSUE THREAD

**Issue by felschr** [tailscale/tailscale#10777]
### What is the issue?

With Mullvad exit node enabled it seems all IPv6 connections stop working. I know that the exit nodes don't support IPv6 yet, but it also doesn't seem to fallback to IPv4 addresses properly.
I noticed this due to hanging git commands.

### Steps to reproduce

```console
> ping gitlab.com # gets stuck without any replies
> ping -4 gitlab.com # works
```

I'm not sure if this is enough or if it requires some specific system configuration.

### OS

Linux

### OS version

NixOS 23.11, Kernel 6.6.8

### Tailscale version

1.56.1

### Other software

networkd, NetworkManager

### Bug report

BUG-df7f0ae5faaa7b04ab154fb0b56a4484d177c1447b39b0c1c8654c68a827ef8e-20240108133244Z-058a757b6ebf700d

**Comment by felschr** [comment]
I think I've recalled something relevant.


I've got the following `/etc/gai.conf` to prioritize IPv6 over IPv4 connections:
```
label ::1/128       0
label ::/0          1
label 2002::/16     2
label ::/96         3
label ::ffff:0:0/96 4
```

This also seems to affect connections through tailscale.
I've briefly removed this file and it does indeed ping the IPv4 address successfully then.

In my OS I had to set this specifically, but I think various other Linux distributions ship with this by default. So this should ideally be something that tailscale can handle.

**Comment by felschr** [comment]
https://www.ietf.org/archive/id/draft-buraglio-6man-rfc6724-update-03.html
https://www.ietf.org/archive/id/draft-buraglio-6man-rfc6724-update-03.html#name-operational-implications

Reading this, it seems the default behavior already prefers IPv6 over IPv4 but only if a GUA is available, because otherwise there can be some issues with NAT including a long connection delay just as I'm experiencing here.

I'll remove the custom `gai.conf` from my system again as the default preference of IPv4 in my case seems to serve a purpose since I don't have a GUA.

**Comment by Aubermean** [comment]
I also have this issue, but there is no custom /etc/gai.conf on my Ubuntu 22.04 new install. #9511

**Comment by bradfitz** [comment]
Dup of #9511. This should be fixed by f205efcf1856c2338989c5e76e2d23fde24fd532 which is in the latest unstable builds and will be in the next stable release probably this week.
